Goal: respect requests for uid-flattening mount options by disabling Unix permissions handling in the kernel driver

Fixes: ?

Status wrt upstream: Forwarded on 2007/05/30

Author: Unknown

Note: Part of no-longer maintained smbfs stuff

Index: samba-3.0.25a/source/client/smbmount.c
===================================================================
--- samba-3.0.25a.orig/source/client/smbmount.c	2007-05-26 07:46:33.884647544 +0200
+++ samba-3.0.25a/source/client/smbmount.c	2007-05-26 07:46:34.272650637 +0200
@@ -213,6 +213,10 @@
   		c->capabilities &= ~CAP_STATUS32;
 		c->force_dos_errors = True;
 	}
+	/* For now, respect requests for uid-flattening mount options
+	   by disabling Unix permissions handling in the kernel driver */
+	if (mount_uid || mount_gid || mount_fmask || mount_dmask)
+		c->capabilities &= ~CAP_UNIX;
 
 	if (!NT_STATUS_IS_OK(cli_session_setup(c, username, 
 					       password, strlen(password),
